pkg/controllers/workloadrebalancer/workloadrebalancer_controller.go

```diff
@@ -220,16 +230,93 @@ func (c *RebalancerController) recordAndCountRebalancerFailed(resource *appsv1al
 
 func (c *RebalancerController) updateWorkloadRebalancerStatus(ctx context.Context, rebalancer *appsv1alpha1.WorkloadRebalancer,
 	newStatus *appsv1alpha1.WorkloadRebalancerStatus) error {
-	rebalancerPatch := client.MergeFrom(rebalancer)
+	if reflect.DeepEqual(rebalancer.Status, newStatus) {
+		return nil
+	}
+
```

Comment:
Here is a bug, `rebalancer.Status` and `newStatus` will always not equal, which is not expected.

1. `rebalancer.Status.FinishTime` may be not nil, but `newStatus` will always be nil, we have not assigned a value to `newStatus.FinishTime`.
2. `rebalancer.Status` is a struct, while `newStatus` is a pointer.

This will lead to: even though `rebalancer.Status` unchanged, we will update `rebalancer.Status.FinishTime` when the rebalancer is reconciled.